src/Nakadi/Client.purs

```diff
@@ -272,3 +274,19 @@ streamSubscriptionEvents sid@(SubscriptionId subId) streamParameters eventHandle
                 )
             ) (err)
   go baseBackOff
+
+getSubscriptionStats
+  :: ∀ r m
+   . MonadAsk (Env r) m
+  => MonadThrow Error m
+  => MonadAff m
+  => SubscriptionId
+  -> m (NakadiResponse (notFound :: E404) (Array SubscriptionEventTypeStats))
+getSubscriptionStats (SubscriptionId subId) = do
+  let url = "/subscriptions/" <> subId <> "/stats"
+  res :: Either Problem {items :: Array SubscriptionEventTypeStats}  <- getRequest url >>= request >>= deserialise
+  stuff <- (res # catchErrors case _ of
+    p @ { status: 401 } -> pure $ lmap e401 res
+    p @ { status: 404 } -> pure $ lmap e404 res
+    p -> unhandled p)
+  pure $ map _.items stuff
```

Comment:
I have no idea how type inference works here. I've tried to do stuff differently but type checker yelled at me.

Reply:
for example this breaks
```purescript
getSubscriptionStats
  :: ∀ r m
   . MonadAsk (Env r) m
  => MonadThrow Error m
  => MonadAff m
  => SubscriptionId
  -> m (NakadiResponse (notFound :: E404) (Array SubscriptionEventTypeStats))
getSubscriptionStats (SubscriptionId subId) = do
  let url = "/subscriptions/" <> subId <> "/stats"
  res :: Either Problem {items :: Array SubscriptionEventTypeStats}  <- getRequest url >>= request >>= deserialise
  let result = map _.items res
  result # catchErrors case _ of
    p @ { status: 401 } -> pure $ lmap e401 res
    p @ { status: 404 } -> pure $ lmap e404 res
    p -> unhandled p
```

Reply:
Simon helped me out. There's `res` hiding in expressions of pattern matching

Reply:
Wow, I did not see this, sorry, and thanks @reactormonk